app/src/org/commcare/activities/FormAndDataSyncer.java

```diff
@@ -115,38 +116,36 @@ public void syncData(final boolean formsToSend, final boolean userTriggeredSync)
             if (userTriggeredSync) {
                 // Remind the user that there's no syncing in demo mode.
                 if (formsToSend) {
-                    activity.displayMessage(Localization.get("main.sync.demo.has.forms"), true, true);
+                    activity.displayBadMessageWithoutToast(Localization.get("main.sync.demo.has.forms"));
                 } else {
-                    activity.displayMessage(Localization.get("main.sync.demo.no.forms"), true, true);
+                    activity.displayBadMessageWithoutToast(Localization.get("main.sync.demo.no.forms"));
                 }
             }
             return;
         }
 
         SharedPreferences prefs = CommCareApplication._().getCurrentApp().getAppPreferences();
-        DataPullTask<CommCareHomeActivity> mDataPullTask = new DataPullTask<CommCareHomeActivity>(
+        DataPullTask<ConnectorWithMessaging> mDataPullTask = new DataPullTask<ConnectorWithMessaging>(
                 u.getUsername(),
                 u.getCachedPwd(),
                 prefs.getString(CommCarePreferences.PREFS_DATA_SERVER_KEY,
                         activity.getString(R.string.ota_restore_url)),
                 activity) {
 
             @Override
-            protected void deliverResult(CommCareHomeActivity receiver, ResultAndError<PullTaskResult> resultAndErrorMessage) {
-                receiver.getUIController().refreshView();
```

Comment:
Pretty sure this isn't needed because the display message calls below trigger a view refresh.